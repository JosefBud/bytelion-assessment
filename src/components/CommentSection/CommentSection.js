// https://jsonplaceholder.typicode.com/comments
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import axios from 'axios';

const CommentSection = () => {
  const [comments, setComments] = useState();
  const [page, setPage] = useState(1);
  const username = useSelector(state => state.username);

  axios.get('https://jsonplaceholder.typicode.com/comments').then(response => {
    setComments(response.data);
  });

  const nextPage = () => {
    setPage(page + 1);
  };

  const prevPage = () => {
    if (page > 1) {
      setPage(page - 1);
    }
  };

  if (username && comments) {
    const slicedComments = comments.slice((page - 1) * 10, page * 10);

    return (
      <div className='commentsection'>
        <h3>Comments:</h3>
        {slicedComments &&
          slicedComments.map((comment, index) => {
            return (
              <div key={'comment' + index} className='commentsection-card'>
                <p className='commentsection-card--name'>
                  <strong>Name:</strong> {comment.name}
                </p>
                <p className='commentsection-card--email'>
                  <strong>E-mail:</strong>{' '}
                  <a href={'mailto: ' + comment.email}>{comment.email}</a>
                </p>
                <p className='commentsection-card--body'>
                  <strong>Message:</strong>
                  <br />
                  {comment.body}
                </p>
              </div>
            );
          })}
        <a onClick={prevPage} href='#'>
          Previous Page
        </a>
        <a onClick={nextPage} href='#'>
          Next Page
        </a>
      </div>
    );
  } else {
    return (
      <div className='commentsection'>
        <h3>Log in to see your comments!</h3>
      </div>
    );
  }
};

export default CommentSection;
