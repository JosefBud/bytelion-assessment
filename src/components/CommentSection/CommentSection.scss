.commentsection {
  text-align: center;
  margin-top: 300px;
  position: relative;

  @media screen and (max-width: 768px) {
    margin-top: 800px;
  }

  &-card {
    text-align: left;
    background: #dedede;
    margin: auto;
    margin-bottom: 10px;
    padding: 5px 15px;
    width: 800px;
    max-width: 100vw;
    border: 1px solid $grey;
    border-radius: 8px;
  }

  a {
    padding: 10px;
  }
}
