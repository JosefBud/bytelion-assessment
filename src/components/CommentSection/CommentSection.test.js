import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import CommentSection from './CommentSection';

describe('<CommentSection />', () => {
  const initialState = {
    username: undefined
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders', () => {
    const render = renderer
      .create(
        <Provider store={store}>
          <CommentSection />
        </Provider>
      )
      .toJSON();

    expect(render).toMatchSnapshot();
  });
});
