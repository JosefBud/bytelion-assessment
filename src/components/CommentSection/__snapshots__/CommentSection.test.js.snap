// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentSection /> renders 1`] = `
<div
  className="commentsection"
>
  <h3>
    Log in to see your comments!
  </h3>
</div>
`;
